src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5211,20 +5535,23 @@ public override BoundNode VisitDynamicObjectCreationExpression(BoundDynamicObjec
         public override BoundNode VisitObjectInitializerExpression(BoundObjectInitializerExpression node)
         {
             // Only reachable from bad expression. Otherwise handled in VisitObjectCreationExpression().
+            // PROTOTYPE(nullable-api): Do we need to analyze child expressions anyway for the public API?
             SetResult(node);
             return null;
         }
 
         public override BoundNode VisitCollectionInitializerExpression(BoundCollectionInitializerExpression node)
         {
             // Only reachable from bad expression. Otherwise handled in VisitObjectCreationExpression().
+            // PROTOTYPE(nullable-api): Do we need to analyze child expressions anyway for the public API?
             SetResult(node);
             return null;
         }
 
         public override BoundNode VisitDynamicCollectionElementInitializer(BoundDynamicCollectionElementInitializer node)
         {
             // Only reachable from bad expression. Otherwise handled in VisitObjectCreationExpression().
+            // PROTOTYPE(nullable-api): Do we need to analyze child expressions anyway for the public API?
```

Comment:
>// PROTOTYPE(nullable-api): Do we need to analyze child expressions anyway for the public API? [](start = 12, length = 94)

For all cases where we're visiting expressions just to populate the map (for instance, `BoundTypeExpression` cases), could we simply have the consumer of the map handle missing expressions with an appropriate default? (The map might be smaller and we wouldn't need to chase down all these errors cases.)

Reply:
There might be some that we can get away with that, but `BoundTypeExpression` isn't one of them due to `var`. We'll need to take the actual inferred type of the variable there and substitute it.

---
In reply to: [260068671](https://github.com/dotnet/roslyn/pull/33591#discussion_r260068671) [](ancestors = 260068671)